server/resource_server/utils/utils.go

```diff
@@ -63,23 +67,22 @@ func CheckPassword(password string, salt string, hash string) bool {
 	return SaltAndHashPassword(password, salt) == hash
 }
 
-func BuildResponse(status bool, message string, data interface{}) Response {
+func BuildResponse(w http.ResponseWriter, statusCode int, message string, data interface{}) Response {
+	w.WriteHeader(statusCode)
 	res := Response{
-		Status:  status,
+		Status:  true,
 		Message: message,
-		Error:   nil,
 		Data:    data,
 	}
 	return res
 }
 
-func BuildErrorResponse(message string, err string, data interface{}) Response {
+func BuildErrorResponse(message string, err string, data interface{}) ErrorResponse {
```

Comment:
do we need to put statusCode in the parameter?

Reply:
Actually we don't need that here becuase if you look at the following function:

```Go
func HandleError(w http.ResponseWriter, status int, message string, err error) {
	w.WriteHeader(status)
	res := BuildErrorResponse(message, err.Error(), EmptyObj{})
	if err := json.NewEncoder(w).Encode(res); err != nil {
		log.Printf("Error sending response: %v", err)
	}
}
```

It writes the status code dynamically here, so either we pass it or not, it does not matter since the `HandleError` function is already doing that, what do you think?

Reply:
got it, looks good